### PR TITLE
Move calls to set_format_arg to Formatter::Common()

### PR DESCRIPTION
**Pull request by Gymnasiast**
This moves all remaining calls to set_format_arg() to use Formatter::Common(). This also means cleaning up many indirect usages of a C-style cast.

### CODE REVIEW

File: src/openrct2-ui/windows/Guest.cpp [hunk 1]
```diff
@@ -1530,17 +1531,21 @@ void window_guest_stats_paint(rct_window* w, rct_drawpixelinfo* dpi)
...
+        ft.Add<uint16_t>(peep->intensity.GetMinimum());
+        ft.Add<uint16_t>(maxIntensity);
         string_id = STR_GUEST_STAT_PREFERRED_INTESITY_BETWEEN;
         if (maxIntensity == 15)
             string_id = STR_GUEST_STAT_PREFERRED_INTESITY_ABOVE;
     }
+    else
+    {
+        ft.Add<uint16_t>(maxIntensity);
+    }
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Guest.cpp:1548]: Alternatively
```
    if (peep->intensity.GetMinimum() != 0)
    {
        ft.Add<uint16_t>(peep->intensity.GetMinimum());
        string_id = STR_GUEST_STAT_PREFERRED_INTESITY_BETWEEN;
        if (maxIntensity == 15)
            string_id = STR_GUEST_STAT_PREFERRED_INTESITY_ABOVE;
    }
    ft.Add<uint16_t>(maxIntensity);
```

(I'd also use a ternary to simplify the `maxIntensity` stuff, but out of scope

File: src/openrct2-ui/windows/Guest.cpp [hunk 2]
```diff
@@ -1553,7 +1558,7 @@ void window_guest_stats_paint(rct_window* w, rct_drawpixelinfo* dpi)
     };
     y += LIST_ROW_HEIGHT;
     int32_t nausea_tolerance = peep->nausea_tolerance & 0x3;
-    set_format_arg(0, rct_string_id, nauseaTolerances[nausea_tolerance]);
+    ft.Add<rct_string_id>(nauseaTolerances[nausea_tolerance]);
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Guest.cpp:1561]:
```suggestion
    auto ft = Formatter::Common();
    ft.Add<rct_string_id>(nauseaTolerances[nausea_tolerance]);
```
To go back to 0

File: src/openrct2-ui/windows/Guest.cpp [hunk 3]
```diff
@@ -1942,58 +1947,67 @@ static rct_string_id window_guest_inventory_format_item(Peep* peep, int32_t item
...
+    ft.Add<rct_string_id>(STR_STRING);
+    ft.Add<const char*>(parkName);
 
     // Special overrides
     Ride* ride{};
     switch (item)
     {
         case SHOP_ITEM_BALLOON:
-            set_format_arg(0, uint32_t, SPRITE_ID_PALETTE_COLOUR_1(peep->BalloonColour) | ShopItems[item].Image);
+            ft.Increment(-ft.NumBytes());
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/Guest.cpp]: Argh this is very hackish X_X
I'm going to think if there's a better way. Seems like here, and on other `case`s you're just trying to rewrite part of what you wrote before the `switch`. I'd save each of these arguments:
```
    ft.Add<uint32_t>(ShopItems[item].Image);
    ft.Add<rct_string_id>(ShopItems[item].Naming.Display);
    ft.Add<rct_string_id>(STR_STRING);
    ft.Add<const char*>(parkName);
```
in a corresponding variable, and update them on the cases, and then actually call:
```
    ft.Add<uint32_t>(ShopItems[item].Image);
    ft.Add<rct_string_id>(ShopItems[item].Naming.Display);
    ft.Add<rct_string_id>(STR_STRING);
    ft.Add<const char*>(parkName);
```
after the switch to get the properly formatted message with decrementing

File: src/openrct2/world/Map.cpp [hunk 4]
```diff
@@ -1254,10 +1254,13 @@ void map_obstruction_set_error_text(TileElement* tileElement, GameActionResult&
             }
             break;
         case TILE_ELEMENT_TYPE_SMALL_SCENERY:
+        {
```
Comment (tupaschoal) [inline src/openrct2/world/Map.cpp:1257]: The changes in this file seems like they belong to another commit

File: src/openrct2/localisation/Localisation.h [hunk 5]
```diff
@@ -112,11 +112,16 @@ class Formatter
         return CurrentBuf;
     }
 
-    void Increment(std::size_t count)
+    void Increment(size_t count)
```
Comment (tupaschoal) [inline src/openrct2/localisation/Localisation.h:115]: Why dropping the `std`?

File: src/openrct2/localisation/Localisation.h [hunk 6]
```diff
@@ -112,11 +112,16 @@ class Formatter
...
     {
         CurrentBuf += count;
     }
 
+    void Rewind()
+    {
+        CurrentBuf -= NumBytes();
+    }
+
```
Comment (tupaschoal) [inline src/openrct2/localisation/Localisation.h:124]: Maybe?
```suggestion
    void Reset()
    {
        CurrentBuf = StartBuf;
    }

```

File: src/openrct2/localisation/Localisation.h (hunk 5 above)
Comment (IntelOrca) [reply src/openrct2/localisation/Localisation.h:115]: Where else do we have `std::size_t`? I think it is just making things consistent.

File: src/openrct2-ui/windows/Guest.cpp (hunk 3 above)
Comment (Gymnasiast) [reply src/openrct2-ui/windows/Guest.cpp]: This had already been changed when you wrote this...

File: src/openrct2/localisation/Localisation.h (hunk 6 above)
Comment (Gymnasiast) [reply src/openrct2/localisation/Localisation.h:124]: Does not compile.

File: src/openrct2/localisation/Localisation.h (hunk 5 above)
Comment (Gymnasiast) [reply src/openrct2/localisation/Localisation.h:115]: I temporarily changed it to a `ssize_t` before opting for another solution. Could be changed back.

File: src/openrct2/world/Map.cpp (hunk 4 above)
Comment (Gymnasiast) [reply src/openrct2/world/Map.cpp:1257]: Er, no. Creating a variable in a `case` requires { and }.

**Comment by duncanspumpkin** [comment]
Rewind seems fine but would prefer we don't use it. For most uses we don't need it but we can fix that eventually.

**Comment by IntelOrca** [comment]
It is a bit counter-intuitive. You basically add arguments to a list, so if you can rewind back, add is no longer inserting them but overwriting old ones. I think each case should just build its own format args rather than build like a default one then overwrite bits of it.

File: src/openrct2/localisation/Localisation.h (hunk 6 above)
Comment (duncanspumpkin) [reply src/openrct2/localisation/Localisation.h:124]: It doesn't compile because the constness of StartBuf is incorrect. It should be a const pointer not a pointer to const data.  `uint8_t* const StartBuf;`. I leave it to a later PR to correct this.

File: src/openrct2/world/Map.cpp (hunk 4 above)
Comment (tupaschoal) [reply src/openrct2/world/Map.cpp:1257]: No no, I meant that the changes you made in this file are not using `Formatter::Common()`, so it's a bit misleading with the PR title and commit messages. I'd just cherry pick the changes in here to another commit and rephrase the PR

File: src/openrct2-ui/windows/Guest.cpp (hunk 3 above)
Comment (tupaschoal) [reply src/openrct2-ui/windows/Guest.cpp]: I did review it commit by commit, maybe that's why

File: src/openrct2/world/Map.cpp (hunk 4 above)
Comment (tupaschoal) [reply src/openrct2/world/Map.cpp:1257]: :(

**Comment by tupaschoal** [comment]
Part of #11389
